### PR TITLE
[ENH] Add drop_constant_columns function 

### LINKED ISSUE #788: [ENH] Function that removes constant columns
Example from @CihanSoylu:

```python
def drop_constant_columns(df):
    """
    Finds and drops the constant columns from a Pandas data frame
    
    Parameters
    ----------
    df : pd.DataFrame
        The Pandas data frame.
    
    Returns
    -------
    pd.DataFrame
        The Pandas data frame with the constant columns dropped.
    """
    # Find the constant columns
    constant_columns = []
    for col in df.columns:
        if len(df[col].unique()) == 1:
            constant_columns.append(col)
    
    # Drop constant columns from df and return it
    return df.drop(labels = constant_columns, 
                   axis = 1
                  )
```

### PR DESCRIPTION
**Pull request by fireddd**
# PR Description

Please describe the changes proposed in the pull request:
- Add drop constant columns function to the functions_list
-
-

**This PR resolves #788 **

# PR Checklist

Please ensure that you have done the following:

1. [x ] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.
 Already a PR in process for this.

3. [x ] Add a line to `CHANGELOG.md` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

# Automatic checks

There will be automatic checks run on the PR. These include:

- Building a preview of the docs on Netlify
- Automatically linting the code
- Making sure the code is documented
- Making sure that all tests are passed
- Making sure that code coverage doesn't go down.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl
- @samukweku

### CODE REVIEW

**Comment by fireddd** [comment]
Hopefully this goes through faster :)

File: janitor/functions.py
Comment (samukweku) [inline janitor/functions.py:4426]: What are the commented out lines for?

Comment (fireddd) [reply janitor/functions.py:4426]: The commented lines below this-> show the example whereas the above show how the function can be called

Comment (ericmjl) [reply janitor/functions.py:4426]: @fireddd @samukweku comments in the code to show examples is a holdover pattern from much before.

Something I learned from _real_ software engineering folks is the following:

1. We have our implementation as a function.
2. Our examples actually live in the test suite, not inside the code.
3. Tutorial-style documentation can reinforce what's in the test suite by bringing them out into the examples gallery.

I can't wait for an in-person sprint at some point to make these a reality and really clean up the codebase. :smile: For now, though, let's leave the comments in-line.

File: tests/conftest.py
Comment (ericmjl) [inline tests/conftest.py:136]:
```suggestion
def df_constant_columns():
    """Return a dataframe that has columns with constant values."""
```

File: tests/functions/test_drop_constant_columns.py
Comment (ericmjl) [inline tests/functions/test_drop_constant_columns.py]: I think I know where this function name came from, @fireddd :smile:, and I think the test function's name isn't right :wink:.

```suggestion
def test_drop_constant_columns(df_constant_columns):
```

Comment (ericmjl) [inline tests/functions/test_drop_constant_columns.py]: As a separate comment, let's also get a docstring for the intent of the test function.

**Review by ericmjl — CHANGES_REQUESTED** [review]
Requesting minor changes from me, @fireddd, and I think we'll be very close to getting this one merged in.

As always, I'll ask for one more pair of eyes on the PR before we decide to merge! (All in the name of maintainability.)

File: janitor/functions.py
Comment (fireddd) [reply janitor/functions.py:4426]: I am not sure what's the correct way here. If there is a better way, i would like to do it that way as it would allow me to learn the best practises. For now, i was just trying to replicate whatever was in the existing code.

File: tests/conftest.py
Comment (fireddd) [reply tests/conftest.py:136]: Thanks for pointing this out.

File: tests/functions/test_drop_constant_columns.py
Comment (fireddd) [reply tests/functions/test_drop_constant_columns.py]: Not my proudest moment :(

Comment (fireddd) [reply tests/functions/test_drop_constant_columns.py]: Ack

**Comment by ericmjl** [comment]
Actually, I just realized @samukweku has done a review already. Okie dokes, we'll hit the merge button when the final suite of tests are done!

**Comment by fireddd** [comment]
> 
> 
> Actually, I just realized @samukweku has done a review already. Okie dokes, we'll hit the merge button when the final suite of tests are done!

noob question here: Aren't all the tests already run?? I can see all checks(10 successful checks) have passed, is there anything else that runs in the background??

**Comment by ericmjl** [comment]
@fireddd nope, nothing else had to run in the bg. As it turns out, GitHub still asked me to trigger the CI just before I posted my comment. I think it's because this was your first official PR to the library. (Your other PRs have triggered the CIs automagically, IIRC.)

**Comment by ericmjl** [comment]
We're good here, thanks for the PR, @fireddd! :smile:

**Pull request merged**
